src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/SummaryListWriter.java

```diff
@@ -163,38 +161,32 @@ protected void generateSummaryListFile(L summaryapi)
      */
     protected void addIndexLink(HtmlId id, String headingKey, Content content) {
         var li = HtmlTree.LI(links.createLink(id,
-                contents.getContent(headingKey)));
+                contents.getContent(headingKey))).setId(HtmlId.of("contents-" + id.name()));
```

Comment:
Suggest adding a comment that `"contents-" + id` appears in the JavaScript code as well. You presumably can't change one without the other.

Reply:
Excellent point.